templates/homepage.html

```diff
@@ -4,36 +4,42 @@
 
 <h2>Create an account</h2>
 
-<form action='/new_user' method='POST'>
+    {% if not session.get('current_user') %}
```

Comment:
I just wanted to show you that you can use the session for authorization on the client, but I know you're implementing the JS parts of the client so you may or may not find this useful.